test/tools/libtesteth/TestOutputHelper.cpp

```diff
@@ -100,3 +108,32 @@ void TestOutputHelper::printTestExecStats()
         saver.restore();
     }
 }
+
+void TestOutputHelper::registerTestFolders(boost::filesystem::path _path)
+{
+    // -t SuiteName/caseName   parse caseName as filter
+    std::string filter;
+    std::string const& tSuiteString = Options::get().rCurrentTestSuite;
+    std::size_t pos = tSuiteString.find('/');
+    if (pos != string::npos)
+        filter = tSuiteString.substr(pos + 1, tSuiteString.length() - pos);
+
+    using fsIterator = boost::filesystem::directory_iterator;
+    for (fsIterator it(_path); it != fsIterator(); ++it)
+    {
+        if (!filter.empty() && it->path().filename() == filter)
```

Comment:
You don't really need to iterate the directory if `filter` is set, right?
What about something like (pseudocode):
```
if (filter.empty())
{
  for (fsIterator it(_path); it != fsIterator(); ++it)
    if (is_directory(it))
      m_testSuiteStatus.emplace(std::make_pair(it->path().filename().c_str(), false));
}
else
{
  if (is_directory(_path + filter))
      m_testSuiteStatus.emplace(std::make_pair(filter, false));
}
    
```